src/app/clusters/ota-requestor/ota-image-processor.h

```diff
@@ -16,32 +16,78 @@
  *    limitations under the License.
  */
 
-/* This file contains the declarations for OTAImageProcessor, a platform-agnostic
- * interface for processing downloaded chunks of OTA image data.
- * Each platform should provide an implementation of this interface.
- */
-
 #pragma once
 
-// This is a platform-agnostic interface for processing downloaded
-// chunks of OTA image data (data could be raw image data meant for flash or
-// metadata). Each platform should provide an implementation of this
-// interface.
-class OTAImageProcessorDriver
+#include <lib/core/CHIPError.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/DLLUtil.h>
+#include <lib/support/Span.h>
+
+namespace chip {
+
+struct OTAImageProcessorParams
+{
+    const char * imageFile;
+    uint64_t downloadedBytes;
+    uint64_t totalFileBytes;
+};
+
+/**
+ * @class OTAImageProcessorInterface
+ *
+ * @brief
+ *   This is a platform-agnostic interface for processing downloaded
+ *   chunks of OTA image data. The data could be raw image data meant for flash or
+ *   metadata. Each platform should provide an implementation of this
+ *   interface.
+ */
+class DLL_EXPORT OTAImageProcessorInterface
 {
 public:
-    // Open file, find block of space in persistent memory, or allocate a buffer, etc.
-    virtual CHIP_ERROR PrepareDownload() = 0;
+    virtual ~OTAImageProcessorInterface() {}
 
-    // Must not be a blocking call to support cases that require IO to elements such as // external peripherals/radios
-    virtual CHIP_ERROR ProcessBlock(ByteSpan & data) = 0;
+    /**
+     * Called to prepare for an OTA image download. This may include but not limited to opening the file, finding a block of space in persistent memory, and allocating a buffer.
+     */
+    virtual CHIP_ERROR PrepareDownload() = 0;
 
-    // Close file, close persistent storage, etc
+    /**
+     * Called when the OTA image download process has completed. This may include but not limited to closing the file and persistent storage.
+     */
     virtual CHIP_ERROR Finalize() = 0;
 
-    virtual chip::Optional<uint8_t> PercentComplete() = 0;
-
-    // Clean up the download which could mean erasing everything that was written,
-    // releasing buffers, etc.
+    /**
+     * Called when the OTA image download process is incomplete or cannot continue. This may include but not limited to erasing everything that has been written and releasing buffers.
+     */
     virtual CHIP_ERROR Abort() = 0;
+
+    /**
+     * Called to process a downloaded block of data. This must not be a blocking call to support cases that require IO to elements such as external peripherals/radios.
```

Comment:
Since it's not blocking, but your example code shows using the CHIP_ERROR, it makes it seem like it's the way to handle errors. The errors would actually need to be notified against the OTARequestor logic, that would then call the ncessary methods on here.